### PR TITLE
IM the new U - new setup procedure

### PR DESCRIPTION
**Pull request by brainuser5705**
IMU setup procedure has been simplified. I tested with BNO055-Sample, and all log messages work as intended. The changes made were referenced from the [documentation](https://cdn-shop.adafruit.com/datasheets/BST_BNO055_DS000_12.pdf) regarding initial states and setup requirements of the device. I added inline comments for more details.

### CODE REVIEW

File: src/BNO055.cpp
Comment (brainuser5705) [inline src/BNO055.cpp:34]: This entire chunk of code (line 35 and above) is not necessary since the device starts at configuration mode by default. It also makes no sense to reset it after it just booted up. In the new code, it first checks if i2c is activated and if so, then it resets the chip, otherwise it waits for i2c to boot up before continuing.

File: src/BNO055.cpp
Comment (brainuser5705) [inline src/BNO055.cpp:77]: A self test is triggered every time the device boots up. It checks if all the sensors and microcontroller of the device is functioning correctly. By checking the results, we can avoid troubleshooting errors.

File: include/BNO055.hpp
Comment (brainuser5705) [inline include/BNO055.hpp:28]: Macro for self-test result register

File: src/BNO055.cpp
Comment (mjmagee991) [inline src/BNO055.cpp]: Update your comments to make it clearer that we're checking if the BNO's I2C is activated, not if the STM's I2C is activated

File: src/BNO055.cpp
Comment (mjmagee991) [inline src/BNO055.cpp]: This is actually just to make sure we are connected to the device. There's no way we're going to have the wrong device connected

File: src/BNO055.cpp
Comment (mjmagee991) [inline src/BNO055.cpp]: Can you reword this? I don't really understand what you're trying to say here

File: src/BNO055.cpp
Comment (mjmagee991) [inline src/BNO055.cpp]: Shouldn't this be `if ((result & 0x0F) != 0x0F)`? That would be how you check that all the least significant bits of the result are 1

File: src/BNO055.cpp
Comment (mjmagee991) [inline src/BNO055.cpp]: Instead of returning a boolean here, you it would be a little cleaner if you defined an enum specific to the BNO that identifies what kind of error occurred

File: src/BNO055.cpp
Comment (mjmagee991) [inline src/BNO055.cpp:80]: What is NDOF?

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Great setup, but there are a couple things that should be changed to make the functionality a bit clearer

File: src/BNO055.cpp
Comment (ActuallyTaylor) [inline src/BNO055.cpp:42]: Do we actually quit initialization here? I don't see a return statement

**Review by ActuallyTaylor — CHANGES_REQUESTED** [review]

File: src/BNO055.cpp
Comment (brainuser5705) [reply src/BNO055.cpp:80]: It is one of the fusion operational modes for BNO055. Here's the excerpt from the datasheet on pg 23:

> 3.3.3.5 NDOF
This is a fusion mode with 9 degrees of freedom where the fused absolute orientation data
is calculated from accelerometer, gyroscope and the magnetometer. The advantages of
combining all three sensors are a fast calculation, resulting in high output data rate, and high
robustness from magnetic field distortions. In this mode the Fast Magnetometer calibration
is turned ON and thereby resulting in quick calibration of the magnetometer and higher
output data accuracy. The current consumption is slightly higher in comparison to the
NDOF_FMC_OFF fusion mode

The original code before my edits had it in NDOF mode.

**Comment by brainuser5705** [comment]
Made the changes to the comments and made an enum for the boolean returns.

File: include/BNO055.hpp
Comment (mjmagee991) [inline include/BNO055.hpp]: This is a nitpick, but generally, we have the successful state have a value of 0, and we call it "OK." It's not really an official standard, but it's pretty common across C code I've seen in my experience

**Review by mjmagee991 — APPROVED** [review]
One nitpick, but otherwise looks great!

**Comment by brainuser5705** [comment]
Awesome, thanks for the review! Do I have to wait for @ActuallyTaylor to approve before I merge? It says that merging is blocked until requested changes are addressed.

**Review by ActuallyTaylor — APPROVED** [review]

**Comment by ActuallyTaylor** [comment]
It looks good @brainuser5705 you should be able to merge it now!

**Pull request merged**
